sql/src/main/java/io/crate/executor/transport/TableCreator.java

```diff
@@ -143,8 +142,8 @@ private void setException(CompletableFuture<Long> result, Throwable e, CreateTab
             // the cause has usually a better more detailed error message
             result.completeExceptionally(e.getCause());
         } else if (statement.ifNotExists() &&
-                   (e instanceof IndexAlreadyExistsException
-                    || (e instanceof IndexTemplateAlreadyExistsException && statement.templateName() != null))) {
+                   (e instanceof ResourceAlreadyExistsException
+                    || (e instanceof IllegalArgumentException && statement.templateName() != null))) {
```

Comment:
not sure if this is really valid, there could be other reasons for an IllegalArgumentException afaik. at least add a FIXME if unsure...